src/main/scala/uk/gov/nationalarchives/DynamoFormatters.scala

```diff
@@ -0,0 +1,151 @@
+package uk.gov.nationalarchives
+
+import cats.data._
+import cats.implicits._
+import org.scanamo._
+import org.scanamo.generic.semiauto._
+
+import java.util.UUID
+import scala.jdk.CollectionConverters._
+import scala.util.Try
+
+object DynamoFormatters {
+
+  val batchId = "batchId"
+  val id = "id"
+  val name = "name"
+  val typeField = "type"
+  val fileSize = "fileSize"
+  val sortOrder = "sortOrder"
+  val parentPath = "parentPath"
+  val title = "title"
+  val description = "description"
+  val checksumSha256 = "checksumSha256"
+  val fileExtension = "fileExtension"
+
+  private type InvalidProperty = (String, DynamoReadError)
+
+  private def stringToType(potentialTypeString: Option[String]): ValidatedNel[InvalidProperty, Type] =
+    potentialTypeString match {
+      case Some("ArchiveFolder") => ArchiveFolder.validNel
+      case Some("ContentFolder") => ContentFolder.validNel
+      case Some("Asset")         => Asset.validNel
+      case Some("File")          => File.validNel
+      case Some(typeString) => (typeField -> TypeCoercionError(new Exception(s"Type $typeString not found"))).invalidNel
+      case None             => (typeField -> MissingProperty).invalidNel
+    }
+
+  implicit val dynamoTableFormat: DynamoFormat[DynamoTable] = new DynamoFormat[DynamoTable] {
+    override def read(dynamoValue: DynamoValue): Either[DynamoReadError, DynamoTable] = {
+      val folderRowAsMap = dynamoValue.toAttributeValue.m().asScala
+
+      def readNumber[T](name: String, toNumberFunction: String => T): ValidatedNel[InvalidProperty, Option[T]] = {
+        folderRowAsMap
+          .get(name)
+          .map { value =>
+            Option(value.n()) match {
+              case Some(value) if Try(toNumberFunction(value)).isSuccess => Option(toNumberFunction(value)).validNel
+              case None => (name -> NoPropertyOfType("Number", DynamoValue.fromAttributeValue(value))).invalidNel
```

Comment:
> Yeah this is when we're expecting the value to be `n()` and it isn't. So there's no property of that type.

Ah, so if the field isn't there, it's a missing property but if the field is there but the value is `null` it's not? But then shouldn't it be `NoValueOfType`?

Reply:
It can either mean the field is null or the  type is something other than `n()` 
There is actually an enum `Type` AWS provide with each of the field types so I'm matching on the attribute type. This way, we know if it's not a number.
If the value is null, we'll end up with `TypeCoercionError` which makes sense.